server_internal_test.go

```diff
@@ -126,3 +126,100 @@ type mockListener struct {
 func (*mockListener) Close() error {
 	return errors.New("mockListener.Close error")
 }
+
+func TestValidateAddr(t *testing.T) {
+	tests := []struct {
+		name            string
+		addr            string
+		expected        string
+		wantErrContains string
+		wantErrIs       error
+	}{
+		{
+			name:     "valid-IPv4-address",
+			addr:     "127.0.0.1:389",
+			expected: "127.0.0.1:389",
+		},
+		{
+			name:     "valid-IPv6-address",
+			addr:     "[::1]:389",
+			expected: "[::1]:389",
+		},
+		{
+			name:     "valid-IPv6",
+			addr:     "2001:db8:3333:4444:5555:6666:7777:8888:389",
+			expected: "2001:db8:3333:4444:5555:6666:7777:8888:389",
+		},
+		{
+			name:     "valid-IPv6-localhost-without-brackets",
+			addr:     "::1:389",
+			expected: "[::1]:389",
+		},
+		{
+			name:     "valid-hostname",
+			addr:     "localhost:389",
+			expected: "localhost:389",
+		},
+		{
+			name:            "err-missing-port-final-colon",
+			addr:            "198.165.1.1:",
+			wantErrContains: "missing port in addr",
+			wantErrIs:       ErrInvalidParameter,
+		},
+		{
+			name:            "missing-port -pv4",
+			addr:            "127.0.0.1",
+			wantErrContains: "missing port in addr",
+			wantErrIs:       ErrInvalidParameter,
+		},
+		{
+			name:            "err-missing-port-ipv6",
+			addr:            "[::1]",
+			wantErrContains: "missing ']' in ipv6 address [::1]",
```

Comment:
This seems supect

Reply:
Oops.  fixed and ty!

Reply:
I refreshed the page but the error message on this test is still saying there is a missing `]` instead of a missing port

Reply:
I need to push the fix.  I'll request re-review when I'm ready.